File: alice/paskills/my_alice/src/main/java/ru/yandex/alice/paskills/my_alice/Application.java
package ru.yandex.alice.paskills.my_alice;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.WebApplicationType;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;

import ru.yandex.web.apphost.api.AppHostService;


@SuppressWarnings("HideUtilityClassConstructor")
@SpringBootApplication(scanBasePackageClasses = Application.class)
public class Application extends SpringBootServletInitializer {

    public static void main(String[] args) {
        var app = new SpringApplication(Application.class);
        // we use AppHostService to serve requests
        app.setWebApplicationType(WebApplicationType.SERVLET);
        app.run(args);
    }

    @ConditionalOnProperty(value = "app.scheduling.enable", havingValue = "true", matchIfMissing = true)
    @Configuration
    @ComponentScan(basePackageClasses = AppHostService.class)
    static class SchedulingConfig {
    }
}

File: alice/paskills/my_alice_pumpkin/src/main/kotlin/ru/yandex/alice/paskills/my_alice/pumpkin/Application.kt
package ru.yandex.alice.paskills.my_alice.pumpkin

import org.springframework.boot.SpringApplication
import org.springframework.boot.WebApplicationType
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.boot.autoconfigure.web.reactive.function.client.ClientHttpConnectorAutoConfiguration
import org.springframework.boot.autoconfigure.web.servlet.error.ErrorMvcAutoConfiguration
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile
import org.springframework.scheduling.annotation.EnableScheduling

@SpringBootApplication(
    scanBasePackageClasses = [ Application::class ],
    exclude = [
        ClientHttpConnectorAutoConfiguration::class,
        ErrorMvcAutoConfiguration::class // remove /error page
    ])
@EnableScheduling
open class Application {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val app = SpringApplication(Application::class.java)
            app.webApplicationType = WebApplicationType.SERVLET
            app.run();
        }
    }
}

@ConditionalOnProperty(value = ["app.scheduling.enable"], havingValue = "true", matchIfMissing = true)
@Configuration
@EnableScheduling
@Profile("!ut")
open class SchedulingConfig
